### PR TITLE
add contract tests

### PR DESCRIPTION
**Pull request by hilary**
https://lobsters.atlassian.net/browse/ENG-15299

This PR contains a new contract testing framework and the contract tests for the address endpoints.

### CODE REVIEW

File: tests/setup.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+};
+
+const specFile = './Lob-API-public.yml';
+
+module.exports.setup = async () => {
+  const operations = await getHttpOperationsFromSpec(specFile);
+  return createClientFromOperations(operations, clientOptions);
+}
+
+const testSecret = 'BASIC dGVzdF8wZGM4ZDUxZTBhY2ZmY2IxODgwZTBmMTljNzliMmY1YjBjYzo='
```
Comment (lbirdeau) [inline tests/setup.js:25]: I think there might be a linter command that looks for keys like this.

**Review by lbirdeau — CHANGES_REQUESTED** [review]
Key

File: tests/setup.js (hunk 1 above)
Comment (lbirdeau) [reply tests/setup.js:25]: nvr mind. looks like this particular endpoint is meant to be public for now. I'll approve.

**Review by lbirdeau — APPROVED** [review]

File: tests/setup.js (hunk 1 above)
Comment (hilary) [reply tests/setup.js:25]: It's the key from the developer docs site, so not adding any additional badness. It's also temporary until we do the ticket that pulls the token from the environment and does the base64 encoding inline. Which is all in the README. :smile:

Comment (hilary) [reply tests/setup.js:25]: Nonetheless, I super appreciate your careful reading and concern for security!

Comment (tlhunter) [inline tests/setup.js:25]: I've never seen basic auth using a spelling other than 'Basic'! I've definitely wrote implementations that check for that exact string. Makes me think of the HtTp header casing.

Comment (tlhunter) [inline tests/setup.js:25]: I've never seen basic auth using a spelling other than 'Basic'! I've definitely wrote implementations that check for that exact string. Makes me think of the HtTp header casing.

**Review by tlhunter — COMMENTED** [review]
LGTM

File: tests/setup.js (hunk 1 above)
Comment (hilary) [reply tests/setup.js:25]: :woman_facepalming: what an excellent point!

**Pull request merged**